pkg/controller/endpointslicemirroring/reconciler.go

```diff
@@ -0,0 +1,291 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpointslicemirroring
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	discovery "k8s.io/api/discovery/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/controller/endpointslicemirroring/metrics"
+	endpointutil "k8s.io/kubernetes/pkg/controller/util/endpoint"
+)
+
+// reconciler is responsible for transforming current EndpointSlice state into
+// desired state
+type reconciler struct {
+	client                clientset.Interface
+	maxEndpointsPerSubset int32
```

Comment:
Please - some comments on things like this would help :)

Reply:
Good call, I had some comments about this elsewhere but completely missed this place. I've added better comments for the surrounding reconciler fields as well.